ui/components/multichain/pages/send/components/quote-card/hooks/useEthFeeData.tsx

```diff
@@ -0,0 +1,65 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { Numeric } from '../../../../../../../../shared/modules/Numeric';
+import {
+  getConversionRate,
+  getGasFeeEstimates,
+  getNativeCurrency,
+} from '../../../../../../../ducks/metamask/metamask';
+import { EtherDenomination } from '../../../../../../../../shared/constants/common';
+import {
+  getCurrentCurrency,
+  checkNetworkAndAccountSupports1559,
+} from '../../../../../../../selectors/selectors';
+import { getUsedSwapsGasPrice } from '../../../../../../../ducks/swaps/swaps';
+import { formatCurrency } from '../../../../../../../helpers/utils/confirm-tx.util';
+import { toFixedNoTrailingZeros } from './utils';
+
+export default function useEthFeeData(gasLimit = 0) {
+  const nativeCurrencySymbol = useSelector(getNativeCurrency);
+
+  const selectedNativeConversionRate = useSelector(getConversionRate);
+
+  const currentCurrency = useSelector(getCurrentCurrency);
+
+  const networkAndAccountSupports1559 = useSelector(
+    checkNetworkAndAccountSupports1559,
+  );
+  const gasPrice = useSelector(getUsedSwapsGasPrice);
```

Comment:
I think this state needs to be populated. Otherwise, when `!networkAndAccountSupports1559`, gas fees will not be computable

Reply:
Copied over some side effect logic to address this.